cpp/src/fil/fil.cu

```diff
@@ -781,19 +787,36 @@ template void init_sparse<sparse_node8>(const raft::handle_t& h, forest_t* pf,
                                         const sparse_node8* nodes,
                                         const forest_params_t* params);
 
+template <typename T, typename L>
 void from_treelite(const raft::handle_t& handle, forest_t* pforest,
-                   ModelHandle model, const treelite_params_t* tl_params) {
+                   const tl::ModelImpl<T, L>& model,
+                   const treelite_params_t* tl_params) {
+  // Invariants on threshold and leaf types
+  static_assert(std::is_same<T, float>::value || std::is_same<T, double>::value,
+                "Model must contain float32 or float64 thresholds for splits");
+  ASSERT((std::is_same<L, float>::value || std::is_same<L, double>::value),
+         "Models with integer leaf output are not yet supported");
+  // Display appropriate warnings when float64 values are being casted into
+  // float32, as FIL only supports inferencing with float32 for the time being
+  if (std::is_same<T, double>::value || std::is_same<L, double>::value) {
+    CUML_LOG_WARN(
+      "The tree ensemble model contains float64 values. Since FIL "
```

Comment:
I might vote for a slightly less verbose warning and more detail in documentation. Maybe something like "float64 values detected in tree ensemble model will be converted to float32. This may reduce model accuracy."? Either way, I think we should rephrase "inaccurate predictions" to something that just indicates reduced accuracy.

Reply:
I specifically chose the phrase "inaccurate prediction" because FIL is an inferencing library. "Reduced accuracy" is vague and may suggest that the model itself is inaccurate, when the truth is that FIL will produce inaccurate prediction due to floating-point truncation.

Reply:
How about "predictions with reduced accuracy"? That makes it clear that the predictions and not the model are the issue.

Reply:
Sounds good. I will also make the warning a little less verbose.